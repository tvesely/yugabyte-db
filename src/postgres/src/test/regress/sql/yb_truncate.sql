CREATE TABLE test_truncate (k int generated by default as identity, v int);

-- test TRUNCATE ... CONTINUE IDENTITY
INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;

TRUNCATE TABLE test_truncate CONTINUE IDENTITY;
SELECT * FROM test_truncate ORDER BY k;

INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;

-- test TRUNCATE ... RESTART IDENTITY
TRUNCATE TABLE test_truncate RESTART IDENTITY;
SELECT * FROM test_truncate ORDER BY k;

INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;

DROP TABLE test_truncate;
