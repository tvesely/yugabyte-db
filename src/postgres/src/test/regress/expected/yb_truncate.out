CREATE TABLE test_truncate (k int generated by default as identity, v int);
-- test TRUNCATE ... CONTINUE IDENTITY
INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;
 k |  v
---+-----
 1 | 101
 2 | 102
 3 | 103
(3 rows)

TRUNCATE TABLE test_truncate CONTINUE IDENTITY;
SELECT * FROM test_truncate ORDER BY k;
 k | v
---+---
(0 rows)

INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;
 k |  v
---+-----
 4 | 101
 5 | 102
 6 | 103
(3 rows)

-- test TRUNCATE ... RESTART IDENTITY
TRUNCATE TABLE test_truncate RESTART IDENTITY;
SELECT * FROM test_truncate ORDER BY k;
 k | v
---+---
(0 rows)

INSERT INTO test_truncate(v) VALUES (101), (102), (103);
SELECT * FROM test_truncate ORDER BY k;
 k |  v
---+-----
 1 | 101
 2 | 102
 3 | 103
(3 rows)

DROP TABLE test_truncate;
